CardBox/Models/ExplodingKittens/ExplodingKittensPlayerPlayCondition.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  ExplodingKittensPlayerPlayCondition.swift
+//  CardBox
+//
+//  Created by mactest on 17/03/2022.
+//
+
+struct ExplodingKittensPlayerPlayCondition: PlayerPlayCondition {
+    let nonActionCards: Set<ExplodingKittensCardType> = [
+        .random1,
+        .random2,
+        .random3
+    ]
+    let actionCards: Set<ExplodingKittensCardType> = [
+        .seeTheFuture,
+        .shuffle,
+        .skip
+    ]
+    var playableCards: Set<ExplodingKittensCardType> {
+        nonActionCards.union(actionCards)
+    }
+
+    func evaluate(gameRunner: GameRunnerReadOnly, args: PlayerPlayConditionArgs) -> Bool {
+        let cards = args.cards
+
+        guard !cards.isEmpty else {
+            return false
+        }
+
+        if cards.count == 1 {
+            let card = cards[0]
+
+            guard let cardType = ExplodingKittensUtils.getCardType(card: card) else {
+                return false
+            }
+            return actionCards.contains(cardType)
+        }
+
+        if cards.count == 2 || cards.count == 3 {
+            return checkSameCards(cards: cards)
+        }
+
+        if cards.count == 5 {
+
+        }
+
+        return false
+    }
+
+    private func checkSameCards(cards: [Card]) -> Bool {
+        let areAllCardsNonAction = cards.allSatisfy({ card in
+            guard let cardType = ExplodingKittensUtils.getCardType(card: card) else {
+                return false
+            }
+            return actionCards.contains(cardType)
+        })
+
+        guard areAllCardsNonAction else {
+            return false
+        }
```

Comment:
Do we need to check if they are non action cards? The rules specified it can be pairs / triplets of action cards as well